packages/dev-tools-pages/ts/components/Animations/index.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+import Lottie from 'react-lottie';
+import styled from 'styled-components';
+
+import { media } from 'ts/variables';
+
+interface AnimationProps {
+    animationData: object;
+    width: number;
+    height: number;
+}
+
+interface AnimationState {
+    width?: number | undefined;
+    height?: number | undefined;
+}
+
+class BaseAnimation extends React.PureComponent<AnimationProps, AnimationState> {
+    public state: AnimationState = {
+        height: undefined,
+        width: undefined,
+    };
+    private _timeout = null as any;
+
+    public componentDidMount(): void {
+        this._updateAnimationSize();
+        window.addEventListener('resize', this._handleResize);
+    }
+
+    public componentWillUnmount(): void {
+        window.removeEventListener('resize', this._handleResize);
+    }
+
+    public render(): React.ReactNode {
+        const { animationData } = this.props;
+        const height = this.state.height || this.props.height;
+        const width = this.state.width || this.props.width;
+
+        return (
+            <Container height={height}>
+                <InnerContainer>
+                    <Lottie
+                        width={width}
+                        height={height}
+                        options={{
+                            loop: true,
+                            autoplay: true,
+                            animationData,
+                        }}
+                    />
+                </InnerContainer>
+            </Container>
+        );
+    }
+
+    private readonly _handleResize = () => {
+        clearTimeout(this._timeout);
+        this._timeout = setTimeout(this._updateAnimationSize, 50);
+    };
+
+    private readonly _updateAnimationSize = () => {
+        const windowWidth = window.innerWidth;
+        let width;
+        let height;
+        if (windowWidth <= 1000) {
+            const maxWidth = windowWidth + 250;
+            const ratio = maxWidth / this.props.width;
+
+            height = Math.round(this.props.height * ratio);
+            width = Math.round(this.props.width * ratio);
+        }
+
+        this.setState({ width, height });
+    };
+}
+
+const Container = styled.div`
+    width: 100%;
+    height: ${(props: { height: number }) => props.height}px;
```

Comment:
You can type the function call.

```
const Container = styled.div<Props>``
```
That prettifies to be kind of ugly but it's what we're doing.